### PR TITLE
[SPARK-20505][ML] Add docs and examples for ml.stat.Correlation and ml.stat.ChiSquareTest.

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?
Add docs and examples for ```ml.stat.Correlation``` and ```ml.stat.ChiSquareTest```.

## How was this patch tested?
Generate docs and run examples manually, successfully.

### CODE REVIEW

**Comment by yanboliang** [comment]
cc @jkbradley

File: examples/src/main/scala/org/apache/spark/examples/ml/ChiSquareTestExample.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      (1.0, Vectors.dense(1.5, 30.0)),
+      (0.0, Vectors.dense(3.5, 30.0)),
+      (0.0, Vectors.dense(3.5, 40.0)),
+      (1.0, Vectors.dense(3.5, 40.0))
+    )
+
+    val df = data.toDF("label", "features")
+    val chi = ChiSquareTest.test(df, "features", "label")
+    val (pValues: Vector, degreesOfFreedom: Array[Int], statistics: Vector) =
+      chi.select("pValues", "degreesOfFreedom", "statistics")
```
Comment (MLnick) [inline examples/src/main/scala/org/apache/spark/examples/ml/ChiSquareTestExample.scala]: Nit: do we need the select here?

File: examples/src/main/python/ml/correlation_example.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  bin/spark-submit examples/src/main/python/ml/correlation_example.py
+"""
+if __name__ == "__main__":
+    spark = SparkSession \
+        .builder \
+        .appName("CorrelationExample") \
+        .getOrCreate()
+
+    # $example on$
+    data = [(Vectors.sparse(4, [(0, 1.0), (3, -2.0)]),),
```
Comment (MLnick) [inline examples/src/main/python/ml/correlation_example.py]: We mix dense and sparse here but not in Scala. Should we make it consistent?

File: examples/src/main/python/ml/chi_square_test_example.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    data = [(0.0, Vectors.dense(0.5, 10.0)),
+            (0.0, Vectors.dense(1.5, 20.0)),
+            (1.0, Vectors.dense(1.5, 30.0)),
+            (0.0, Vectors.dense(3.5, 30.0)),
+            (0.0, Vectors.dense(3.5, 40.0)),
+            (1.0, Vectors.dense(3.5, 40.0))]
+    df = spark.createDataFrame(data, ["label", "features"])
+
+    r = ChiSquareTest.test(df, "features", "label").head()
+    print("pValues: " + str(r[0]))
```
Comment (MLnick) [inline examples/src/main/python/ml/chi_square_test_example.py]: we could do `r.pValues`, `r. degreesOfFreedom`, `r. statistics`?

**Review by MLnick — COMMENTED** [review]
Minor nits - otherwise LGTM

File: examples/src/main/python/ml/correlation_example.py (hunk 2 above)
Comment (yanboliang) [inline examples/src/main/python/ml/correlation_example.py]: Done.

File: examples/src/main/python/ml/chi_square_test_example.py (hunk 3 above)
Comment (yanboliang) [inline examples/src/main/python/ml/chi_square_test_example.py]: Done.

File: examples/src/main/scala/org/apache/spark/examples/ml/ChiSquareTestExample.scala (hunk 1 above)
Comment (yanboliang) [inline examples/src/main/scala/org/apache/spark/examples/ml/ChiSquareTestExample.scala]: Removed.

**Comment by yanboliang** [comment]
Merged into master and branch-2.2. Thanks for reviewing.

**Pull request closed without merging**